src/controllers/session.ts

```diff
@@ -0,0 +1,60 @@
+import { Response, NextFunction } from 'express';
+import { RequestWithUserRole, AppointmentWithTherapistOptional } from '../types';
+import generateMeeting from '../services/calendar';
+import { templateErrors } from '../helpers';
+import { BookedAppointment } from '../services';
+import { User } from '../models';
+
+const createSessionController = async (
+  req: RequestWithUserRole,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const userData = req.user;
+    const { appointmentId } = req.body;
+
+    const appointment: AppointmentWithTherapistOptional
+    | null = await BookedAppointment(appointmentId);
+
+    if (!appointment) {
+      throw templateErrors.BAD_REQUEST('The requested appointment is not available.');
+    }
+    if (!userData) {
+      throw templateErrors.UNAUTHORIZED('user not found');
```

Comment:
I don't understand why you check if user data is not found since we already check for it in the checkAuth middleware before entering this route